tests/unit/local/lib/test_sam_api_provider.py

```diff
@@ -0,0 +1,192 @@
+from unittest import TestCase
+
+from samcli.commands.local.lib.provider import Api
+from samcli.commands.local.lib.sam_api_provider import SamApiProvider
+
+
+class TestSamStageValues(TestCase):
```

Comment:
These tests should be in `tests/unit/commands/local/lib/test_sam_api_provider.py`. These tests are validating the whole api is created. This is ok but if you keep it this way, combine it with some other tests that verify the api as a whole. The other way to write these would be to only validate stage variables and stage name in there own tests. I am personally a fan of individual tests that only assert one thing but ok with either way.